calamari_ocr/ocr/datasets/pagexml_dataset/dataset.py

```diff
@@ -76,14 +76,22 @@ def _samples_gt_from_book(self, root, img,
 
         img_w = int(root.xpath('//ns:Page',
                                namespaces=ns)[0].attrib["imageWidth"])
-        tequivs = root.xpath('//ns:TextEquiv[@index="{}"]'.format(self.text_index),
-                             namespaces=ns)
-        for l in tequivs:
-            parat = l.getparent().attrib
+        textlines = root.xpath('//ns:TextLine', namespaces=ns)
+
+        for textline in textlines:
+            tequivs = textline.xpath('./ns:TextEquiv[@index="{}"]'.format(self.text_index),
+                                namespaces=ns)
+            if len(tequivs) > 1:
+                raise IOError('TextLine includese TextEquivs with non unique ids')
```

Comment:
typo
```python
import logging
logger = logging.getLogger(__name__)

# instead of raise log a warning
logger.warning("..... PageXML is inavalid"
```